src/Kmd.Logic.Gateway.Automation/Publish.cs

```diff
@@ -232,6 +232,36 @@ private async Task CreateApi(IGatewayClient client, Guid subscriptionId, Guid pr
             {
                 existingApis.Add(createdApi);
                 this.publishResults.Add(new GatewayAutomationResult() { ResultCode = apiVersionSetId.HasValue ? ResultCode.VersionCreated : ResultCode.ApiCreated, EntityId = createdApi.Id });
+                await this.CreateRevisions(client, subscriptionId, createdApi.Id.Value, folderPath, apiVersion.Revisions).ConfigureAwait(false);
+            }
+        }
+
+        private async Task CreateRevisions(IGatewayClient client, Guid subscriptionId, Guid apiVersionId, string folderPath, IEnumerable<Revision> revisions)
+        {
+            if (revisions == null)
+            {
+                return;
+            }
+
+            foreach (var revision in revisions)
+            {
+                await this.CreateRevision(client, subscriptionId, apiVersionId, folderPath, revision).ConfigureAwait(false);
+            }
+        }
+
+        private async Task CreateRevision(IGatewayClient client, Guid subscriptionId, Guid apiVersionId, string folderPath, Revision revision)
+        {
+            using var revisionOpenApiSpec = new FileStream(path: Path.Combine(folderPath, revision.OpenApiSpecFile), FileMode.Open);
+            var revisionResponse = await client.CreateRevisionAsync(
+                subscriptionId: subscriptionId,
+                apiId: apiVersionId,
```

Comment:
Are you sure that `apiVersionId` should be assigned to `apiId`?

Reply:
yes apiversionId used for revision

Reply:
But it shouldn't be called apiVersionId, but rather apiId.